oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -191,7 +197,10 @@ public final Collection<String> getScopes() {
   }
 
   /**
-   * If scopes is specified, add "?scopes=comma-separated-list-of-scopes" to the token url.
+   * If scopes is specified, add "?scopes=comma-separated-list-of-scopes" to the token url. If
+   * transport is specified, add "?transport=xyz" to the token url; xyz is one of "alts" or "mtls".
+   * If bindingEnforcement is specified, add "?binding-enforcement=xyz" to the token url; xyz is one
+   * of "iam-policy" or "on".
```

Comment:
If these have a limited set of options, would it be possible we have a Transport Enum and a BindingEnforcement enum?

The transport setter would take in a set of Transport enums to use.

Reply:
> If these have a limited set of options
Yes there are a limited set of options. Transport can be "alts" or "mtls". BindingEnforcement can be "on" or "iam-policy"

Thanks for suggesting this @lqiu96. This is something that we can do. It would require exposing a public `Transport` enum and `BindingEnforcement` enum so that the logic in sdk-platform-java can set these values. Are you ok with adding these public enums to the auth library?

Reply:
> It would require exposing a public Transport enum and BindingEnforcement enum so that the logic in sdk-platform-java can set these values. Are you ok with adding these public enums to the auth library?

Ah I missed this. I think that's fine and would be better to cover the combination of transport and bindingEnforcement configurations. The invalid configurations between the two enums can be handled by the server side logic, but at least this would help users choose the correct configurations.

Just to confirm, these enums aren't expect to change much (or at all)

Reply:
>  I think that's fine and would be better to cover the combination of transport and bindingEnforcement configurations. The invalid configurations between the two enums can be handled by the server side logic, but at least this would help users choose the correct configurations.

SG

> Just to confirm, these enums aren't expect to change much (or at all)

Yes, we don't expect them to change, as the design has been finalized and approved by all stakeholders, and implementation has begun. When this feature is non-experimental, the enums would be confirmed to not change.

@lqiu96 in the latest commit, I added these enums.